services/headless-lms/server/src/controllers/course_material/courses.rs

```diff
@@ -664,6 +665,49 @@ async fn get_page_by_course_id_and_language_group(
     token.authorized_ok(web::Json(page))
 }
 
+/**
+POST `/api/v0/{course_id}/course-instances/{course_instance_id}/student-countries/{country_code}` - Add a new student's country entry.
+*/
+#[generated_doc]
+#[instrument(skip(pool))]
+async fn student_country(
+    query: web::Path<(Uuid, Uuid, String)>,
+    pool: web::Data<PgPool>,
+    user: AuthUser,
+) -> ControllerResult<()> {
+    let mut conn = pool.acquire().await?;
+    let (course_id, course_instance_id, country_code) = query.into_inner();
+
+    models::student_countries::insert(
+        &mut conn,
+        user.id,
+        course_id,
+        course_instance_id,
+        &country_code,
+    )
+    .await?;
+    let token = skip_authorize()?;
+
+    token.authorized_ok(())
+}
+
+/**
+GET `/api/v0/{course_id}/student-countries - Returns countries of student registered in a course.
+ */
+#[generated_doc]
+#[instrument(skip(pool))]
+async fn get_student_countries(
+    course_id: web::Path<Uuid>,
+    pool: web::Data<PgPool>,
+    user: AuthUser,
+) -> ControllerResult<web::Json<Vec<StudentCountry>>> {
```

Comment:
Make this return a hashmap where the countries are the keys and the values are the counts of students in that country.  Also, make another endpoint where one can get their own answer to the form back.

{
 "en": 2,
"fi": 3,
}